awscli/argprocess.py

```diff
@@ -316,171 +344,32 @@ def get_parse_method_for_param(self, cli_argument, value=None):
-        return parsed
-
-    def _special_key_value_parse(self, param, value):
-        # This is a special key value parse that can do the normal
-        # key=value parsing, *but* supports a few additional conveniences
-        # when working with a structure with a single element.
-        # Precondition: param is a shape of structure(scalar)
-        if self._is_special_case_key_value(param, value):
-            # We have an even shorter shorthand syntax for structure
-            # of scalars of a single element with a member name of
-            # 'Value'.
-            return {'Value': value}
-        else:
-            return self._key_value_parse(param, value)
-
-    def _is_special_case_key_value(self, param, value):
-        members = param.members
-        if len(param.members) == 1:
-            if list(members.keys())[0] == 'Value' and \
-                    '=' not in value:
-                return True
-        return False
-
-    def _key_value_parse(self, param, value):
-        # The expected structure is:
-        #  key=value,key2=value
-        # that is, csv key value pairs, where the key and values
-        # are separated by '='.  All of this should be whitespace
-        # insensitive.
-        parsed = OrderedDict()
-        parts = self._split_on_commas(value)
-        valid_names = self._create_name_to_params(param)
-        for part in parts:
-            try:
-                key, value = part.split('=', 1)
-            except ValueError:
-                raise ParamSyntaxError(part)
-            key = key.strip()
-            value = value.strip()
-            if valid_names and key not in valid_names:
-                raise ParamUnknownKeyError(key, valid_names)
-            if valid_names:
-                sub_param = valid_names[key]
-                if sub_param is not None:
-                    # TODO: you are here.  unpack_scalar_cli_arg takes
-                    # the cli_name, but we don't have it.  What are our
-                    # options?
-                    value = unpack_scalar_cli_arg(sub_param, value)
-            parsed[key] = value
-        return parsed
-
-    def _create_name_to_params(self, param):
-        if param.type_name == 'structure':
-            return dict([(member_name, p) for member_name, p
-                         in param.members.items()])
-        elif param.type_name == 'map' and hasattr(param.key, 'enum'):
-            return dict([(v, None) for v in param.key.enum])
-
-    def _split_on_commas(self, value):
-        try:
-            return utils.split_on_commas(value)
-        except ValueError as e:
-            raise ParamSyntaxError(six.text_type(e))
+            return False
+        model = cli_argument.argument_model
+        # The second case is to make sure the argument is sufficiently
+        # complex, that is, it's base type is a complex type *and*
+        # if it's a list, then it can't be a list of scalar types.
```

Comment:
That answers it for me. I was just making sure that you ignored the list of scalars, in terms of the shorthand parser, because it is handled in the line that I linked.